src/ExtendedXmlSerializer/Cache/ObjectAccessors.cs

```diff
@@ -111,21 +111,21 @@ internal static AddItemToDictionary CreateMethodAddToDictionary(Type type)
             return compiled;
         }
 
-        internal static AddItemToCollection CreateMethodAddCollection(Type type)
+        internal static AddItemToCollection CreateMethodAddCollection(Type type, Type elementType = null)
         {
             // Object (type object) from witch the data are retrieved
             ParameterExpression itemObject = Expression.Parameter(typeof(object), "item");
 
             // Object casted to specific type using the operator "as".
             UnaryExpression itemCasted = Expression.Convert(itemObject, type);
 
-            var arguments = type.GetGenericArguments();
+            var parameterType = elementType ?? type.GetGenericArguments()[0];
 
             ParameterExpression value = Expression.Parameter(typeof(object), "value");
 
-            Expression castedParam = Expression.Convert(value, arguments[0]);
+            Expression castedParam = Expression.Convert(value, parameterType);
 
-            MethodInfo method = type.GetMethod("Add");
+            MethodInfo method = AddMethodLocator.Default.Locate( type );
```

Comment:
Ah yes... consider the case of `IList<T>`.  It does not have an `Add` method.  So what this does is search through its interfaces and finds it.  This is also the case for classes that explicitly implement an interface:

```
void System.Collections.Generic.ICollection<object>.Add(object item)
{
//...
}
```